dvc/repo/experiments/executor/base.py

```diff
@@ -450,9 +452,32 @@ def _repro_args(cls, dvc):
             kwargs = {}
         return args, kwargs
 
+    @staticmethod
+    def _auto_push(git_remote: str, dvc: "Repo", scm: "Git"):
+        from dvc.repo.experiments.push import push
+
+        if git_remote == scm.root_dir:
+            logger.warning(
+                f"{DVC_EXP_AUTO_PUSH} {git_remote} is the running "
+                "repository auto push will not work"
+            )
+            return
+
+        branch = scm.get_ref(EXEC_BRANCH, follow=False)
+        branch_name = ExpRefInfo.from_ref(branch).name
+        push(
+            dvc,
+            git_remote,
+            branch_name,
```

Comment:
It would be better to call `experiments.push` with the full refname string returned by `scm.get_ref()` here, instead of using this `branch_name` (which is the shortened exp name). Otherwise we will end up doing the entire shortname to full ref string lookup/conversion again in `experiments.push`.